### PR TITLE
Split up Using.scala into 2.11 and 2.12 versions

### PR DESCRIPTION
**Pull request by kwalcock**
This is so that a Source can be automatically released in 2.11

### CODE REVIEW

File: compat/src/main/scala-2.11/scala/util/Using.scala [hunk 1]
```diff
@@ -0,0 +1,409 @@
...
+    implicit object AutoCloseableIsReleasable extends Releasable[AutoCloseable] {
+      def release(resource: AutoCloseable): Unit = resource.close()
+    }
+
+    /** An implicit `Releasable` for [[scala.io.Source `Source`s]] which aren't [[java.lang.AutoCloseable `AutoCloseable`s]] in Scala 2.11. */
+    implicit object SourceReleasable extends Releasable[Source] {
+      def release(resource: Source): Unit = resource.close()
+    }
+  }
+
```
Comment (preveen-stack) [inline compat/src/main/scala-2.11/scala/util/Using.scala:408]: guess we can remove this empty line

Comment (kwalcock) [reply compat/src/main/scala-2.11/scala/util/Using.scala:408]: It was there before, so I left it in.  Can it be configured in scalafmt if it is a problem?

Comment (SethTisue) [reply compat/src/main/scala-2.11/scala/util/Using.scala:408]: it's fine to clean up formatting in the course of making changes, but it's also fine not to

**Comment by SethTisue** [comment]
Thank you!

**Pull request merged**